mlflow/store/tracking/databricks_rest_store.py

```diff
@@ -678,25 +701,36 @@ def link_traces_to_run(self, trace_ids: list[str], run_id: str) -> None:
         if not trace_ids:
             return
 
-        # Group traces by location (None for V3 traces)
-        # For V3 traces, store original IDs; for V4 traces, store OTEL trace IDs
-        v3_trace_ids: list[str] = []
-        v4_traces_by_location: dict[str, list[str]] = defaultdict(list)
+        traces_by_location = self._group_traces_by_location(trace_ids)
 
-        for trace_id in trace_ids:
-            location_id, otel_trace_id = parse_trace_id_v4(trace_id)
+        for location_id, batch_trace_ids in traces_by_location.items():
             if location_id is None:
-                v3_trace_ids.append(trace_id)
+                super().link_traces_to_run(batch_trace_ids, run_id)
             else:
-                v4_traces_by_location[location_id].append(otel_trace_id)
+                self._batch_link_traces_to_run(location_id, batch_trace_ids, run_id)
+
+    def unlink_traces_from_run(self, trace_ids: list[str], run_id: str) -> None:
+        """
+        Unlink multiple traces from a run by removing trace-to-run relationships.
 
-        # Link V3 traces
-        if v3_trace_ids:
-            super().link_traces_to_run(v3_trace_ids, run_id)
+        Args:
+            trace_ids: List of trace IDs to unlink from the run.
+            run_id: ID of the run to unlink traces from.
+        """
+        if not trace_ids:
+            return
+
+        traces_by_location = self._group_traces_by_location(trace_ids)
+
+        if v3_trace_ids := traces_by_location.pop(None, []):
+            raise MlflowException(
+                "Unlinking V3 traces from runs is not supported. Only V4 traces (with UC "
```

Comment:
I think we shouldn't mention v3 or v4 in the error message 🤔 Probably just mention we support traces with UC schema locations